File: archive/j/java/Rot13.java
public class Rot13
{
   public static void main(String[] args)
   {
      if(args.length<1) {
         System.out.println("Usage: please provide a string to encrypt");
      }
      else {
         String code = args[0];
         String result = "";
         if(code.length()==0) {
            System.out.println("Usage: please provide a string to encrypt");
         }
         else {
            String lower = "abcdefghijklmnopqrstuvwxyz";
            String upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
            for(int a=0; a<code.length(); a++) {
               String ch = code.substring(a,a+1);
               int l = lower.indexOf(ch);
               int u = upper.indexOf(ch);
               if(l!=-1) {
                  result+=lower.substring( (l+13)%26,(l+14)%26!=0 ? (l+14)%26 : l+14 );
               }
               else if(u!=-1) {
                  result+=upper.substring( (u+13)%26,(u+14)%26!=0 ? (u+14)%26 : u+14 );
               }
               else {
                  result+=ch;
               }
            }
            System.out.println(result);
         }
      }
   }
}

File: archive/t/typescript/rot13.ts
function rot13(input: string): string {
  if (!input) {
      return "Usage: please provide a string to encrypt";
  }

  return input.split('').map(char => {
      const code = char.charCodeAt(0);
      // Check if the character is an uppercase letter
      if (code >= 65 && code <= 90) {
          return String.fromCharCode((code - 65 + 13) % 26 + 65);
      }
      // Check if the character is a lowercase letter
      else if (code >= 97 && code <= 122) {
          return String.fromCharCode((code - 97 + 13) % 26 + 97);
      }
      // If it's not a letter, return the character unchanged
      return char;
  }).join('');
}

const args = process.argv.slice(2);

if (args.length !== 1) {
  console.log("Usage: please provide a string to encrypt");
  process.exit(1);
} else {
  const input = args[0];
  const result = rot13(input);
  console.log(result);
}
